lib/vdsm/storage/mpathconf.py

```diff
@@ -432,6 +435,58 @@ def read_metadata():
     return Metadata(REVISION_MISSING, private)
 
 
+def check_local_config():
+    """
+    Check if 'user_friendly_names' is enabled in the multipathd configuration.
+    The configuration is obtained by running the multipathd command to show
+    the local configuration and capturing the output.
+    Parse the output and check that the contents are valid.
+
+    Returns:
+        vdsm.storage.mpathconf.Result
+    """
+    error = None
+    try:
+        mpath_conf = _parse_conf(io.StringIO(multipath.show_config_local()))
+        issues = _check_conf(mpath_conf)
+    except cmdutils.Error as e:
+        error = f"Cannot check multipath configuration: {e}"
+        issues = []
+    else:
+        if issues:
+            error = ("Invalid configuration: 'user_friendly_names' is "
+                     "enabled in multipath configuration")
+    return Result(error, issues)
+
+
+def _check_conf(mpath_conf):
+    """
+    Check that 'user_friendly_names' is disabled in all the children
+    of the parsed multipathd configuration.
+    Search is recursive.
+
+    Args:
+        mpath_conf: Parsed structure of the multipathd configuration output.
+
+    Returns:
+        List[Section]: Sections that have invalid configuration.
+    """
+    issues = []
+    for item in mpath_conf:
+        if not item.children or item.name == "blacklist":
```

Comment:
Three is also blacklist_exceptions which can contain the same content as blacklist.

Reply:
True, I just checked the documentation. I have added both. The other sections are "defaults", "multipaths", and "devices", so we should be fine excluding those two.